riscv-asm.md

```diff
@@ -184,6 +184,150 @@ Directive    | Arguments                      | Description
+.L3:
+    addi    a1,a1,1
+    addi    a5,a5,1
+    lbu     a4,-1(a1)
+    sb      a4,-1(a5)
+    bne     a5,a2,.L3
+.L2:
+    ret
+
+.option push     # Push current option to stack.
+.option arch, +v # Enable vector extension, we can use any instruction in imafdcv extension.
+memcpy_vec:
+    mv a3, a0
+.Lloop:
+    vsetvli t0, a2, e8, m8, ta, ma
+    vle8.v v0, (a1)
+    add a1, a1, t0
+    sub a2, a2, t0
+    vse8.v v0, (a3)
+    add a3, a3, t0
+    bnez a2, .Lloop
+    ret
+.option pop   # Pop current option from stack, restore the enabled ISA extension status to imafdc.
+
+.option push     # Push current option to stack.
+.option arch, -c # Disable compressed extension, we can't use any instruction in extension.
+memcpy_norvc:
+    add     a5,a1,a2
+    beq     a1,a5,.L2
+    add     a2,a0,a2
+    mv      a5,a0
+.L3:
+    addi    a1,a1,1
+    addi    a5,a5,1
+    lbu     a4,-1(a1)
+    sb      a4,-1(a5)
+    bne     a5,a2,.L3
+.L2:
+    ret
+.option pop   # Pop current option from stack, restore the enabled ISA extension status to imafdc.
+
+.option push           # Push current option to stack.
+.option arch, =rv64imc # Set arch to rv64imc.
+    nop
+.option pop   # Pop current option from stack, restore the enabled ISA extension status to imafdc.
+
+```
+
+NOTE: A typical use case is is with `ifunc`, e.g. the libc is built with
+`rv64gc`, but a few functions like memcpy provide two versions, one built with
+`rv64gc` and one built with `rv64gcv`, and then select between them by ifunc
+mechanism at run-time.  However we don't want to change the minimal execution
+environment requirement to `rv64gcv`, since the `rv64gcv` version will be
+invoked only if the execution environment supports the vector extension, so
+the minimal execution environment requirement still is `rv64gc`.
+
+NOTE: `.option arch, +` will also enable all required extensions, for example,
+`rv32i` + `.option arch, +v` will also enable `f`, `d`, `zve32x`, `zve32f`,
+`zve64x`, `zve64f`, `zve64d`, `zvl32b`, `zvl64b` and `zvl128b` extensions.
+
+NOTE: We recommended `.option arch, +` and `.option arch, -` are used with
+`.option push`/`.option pop` instead of a `.option arch, +` / `.option arch, -`
+pair, because `.option arch, +` will enable all required extensions, but
+`.option arch, -` only disable specific extension, so the result might be
+unexpected, for example: `rv32i` + `.option arch, +v` + `.option arch, -v`
+will result `rv32ifd_zve32x_zve32f_zve64x_zve64f_zve64d_zvl32b_zvl64b_zvl128b`
+not `rv32i`.
+
+#### `pic`/`nopic`
+
+Set the code model to PIC (position independent code) or non-PIC. This will
+affect the expansion of the `la` pseudoinstruction, refer to
+[listing of standard RISC-V pseudoinstructions](#pseudoinstructions).
+
+#### `relax`/`norelax`
+
+Enable/disable linker relaxation for the following code region.
+
+NOTE: Code region follows by `.option relax` will emit
+`R_RISCV_RELAX`/`R_RISCV_ALIGN` even linker unsupport relaxation, suggested
```

Comment:
"even linker unsupport relaxation" -> "even if the linker does not support relaxation"?